### PR TITLE
docs: Add SyncBatchNorm example to custom trainer guide

### PR DESCRIPTION
**Pull request by artest08**
Shows how to subclass RTDETRTrainer to convert BatchNorm to SyncBatchNorm before DDP wraps the model, useful for multi-GPU fine-tuning with small per-GPU batches.

Adds a `## Synchronized BatchNorm for Multi-GPU Training` section to the custom trainer guide showing how to subclass `RTDETRTrainer` and override `set_model_attributes()` to convert BatchNorm to SyncBatchNorm before DDP wraps the model. The conversion is gated by a `world_size > 1` guard so the same trainer is safe in single-GPU runs, and the change is docs-only with no library or `default.yaml` modifications.

```python
from torch import nn

from ultralytics import RTDETR
from ultralytics.models.rtdetr.train import RTDETRTrainer


class SyncBNTrainer(RTDETRTrainer):
    """RT-DETR trainer with SyncBatchNorm for multi-GPU training."""

    def set_model_attributes(self):
        super().set_model_attributes()
        if self.world_size > 1:
            self.model = nn.SyncBatchNorm.convert_sync_batchnorm(self.model)


RTDETR("rtdetr-l.pt").train(data="coco8.yaml", epochs=20, device=[0, 1], trainer=SyncBNTrainer)
```

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
📚 This PR expands the custom trainer guide by adding a new example for using `SyncBatchNorm` in multi-GPU training, helping users improve training stability in small-batch distributed setups.

### 📊 Key Changes
- Added a **sixth customization example** to the [custom trainer guide](https://github.com/ultralytics/ultralytics/blob/main/docs/en/guides/custom-trainer.md): **Synchronized BatchNorm for Multi-GPU Training** 🚀
- Updated the guide’s **description and keywords** to include `SyncBatchNorm` and **multi-GPU training** for better discoverability 🔍
- Documented **when and why** to use `SyncBatchNorm`, especially for **RT-DETR fine-tuning** and other cases with **small per-GPU batch sizes**
- Included a practical code example showing how to:
  - subclass `RTDETRTrainer`
  - override `set_model_attributes()`
  - convert `BatchNorm` layers to `SyncBatchNorm` only when `world_size > 1` 🛠️
- Added a simple recommendation table covering:
  - small-batch multi-GPU training ✅
  - large-batch multi-GPU training ⚖️
  - single-GPU training 🚫

### 🎯 Purpose & Impact
- Helps users better understand **how to customize training for distributed multi-GPU workloads** 🤝
- Improves documentation for **advanced training setups**, especially for users fine-tuning RT-DETR or similar models with limited batch size per GPU
- Gives a **safe, reusable implementation pattern** that avoids applying `SyncBatchNorm` in single-GPU runs
- Can lead to **more stable convergence and better training behavior** in the right scenarios, with the tradeoff of a small communication overhead 📈
- Makes the custom trainer docs more complete and practical for both **researchers and production users** using Ultralytics training workflows ✨

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @artest08, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help streamline review; an engineer will assist you shortly. Since this PR updates documentation for the custom trainer guide with a `SyncBatchNorm` multi-GPU example, please review the checklist below to help ensure a smooth integration ✨

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean documentation update. The new SyncBatchNorm section is consistent with the existing guide structure, and the example appears accurate and actionable for multi-GPU custom trainer usage.

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉

> “The best way to predict the future is to create it.” — Peter Drucker

This update does exactly that by making our custom trainer docs more practical for real-world multi-GPU workflows. Adding a clear `SyncBatchNorm` example for small-batch distributed training is a valuable improvement that will help users fine-tuning RT-DETR and other advanced models train more confidently and effectively.

Big thanks to @artest08 for the thoughtful documentation contribution, and to @glenn-jocher for helping move it across the finish line. This is a great example of how small, targeted doc improvements can have a big impact for the community.
